packages/block-editor/src/hooks/test/anchor.js

```diff
@@ -39,6 +39,22 @@ describe( 'anchor', () => {
 
 			expect( settings.attributes ).toHaveProperty( 'anchor' );
 		} );
+
+		it( 'should not override attributes defined in settings', () => {
+			const settings = registerBlockType( {
+				...blockSettings,
+				supports: {
+					anchor: true,
+				},
+				attributes: {
+					anchor: {
+						type: 'string',
+					},
+				},
+			} );
+
+			expect( settings.attributes.anchor ).not.toHaveProperty( 'source' );
```

Comment:
The following might be safer to use:
```suggestion
			expect( settings.attributes.anchor ).toEqual( { type: 'string' } );
```

adding `default` with some unique value in the definition of `attributes` would make it more obvious that it doesn't change.

Reply:
Done, thanks for the suggestion!